query-core/src/main/java/org/infinispan/query/core/impl/QueryCache.java

```diff
@@ -68,16 +70,28 @@ public <T> T get(String cacheName, String queryString, List<FieldAccumulator> ac
 
    public void clear() {
       log.debug("Clearing query cache for all caches");
-      getCache().clear();
+      if (cache != null) {
+         cache.clear();
+      }
    }
 
    public void clear(String cacheName) {
       log.debugf("Clearing query cache for cache %s", cacheName);
-      getCache().keySet().removeIf(k -> k.cacheName.equals(cacheName));
+      if (cache != null) {
+         cache.keySet().removeIf(k -> k.cacheName.equals(cacheName));
+      }
    }
 
    private Cache<QueryCacheKey, Object> getCache() {
-      return cacheManager.getCache(QUERY_CACHE_NAME);
+      if (cache != null) {
+         return cache;
+      }
+      synchronized (this) {
+         if (cache == null) {
+            cache = cacheManager.getCache(QUERY_CACHE_NAME);
+         }
+      }
+      return cache;
```

Comment:
nitpicking, you don't need the synchronized block because `cacheManager.getCache(QUERY_CACHE_NAME)` always returns the same instance. 

```suggestion
      if (cache == null) {
         cache = cacheManager.getCache(QUERY_CACHE_NAME);
      }
      return cache;
```

Reply:
great! I'll apply the change. thank you